modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -867,10 +867,6 @@ private void updateCodegenParametersEnum(List<CodegenParameter> parameters, List
             if (!parameter.isContainer && nullableType.contains(parameter.dataType)) {
                 parameter.vendorExtensions.put("x-csharp-value-type", true);
             }
-
-            if (!parameter.required && parameter.vendorExtensions.get("x-csharp-value-type") != null) { //optional
-                parameter.dataType = parameter.dataType + "?";
-            }
```

Comment:
why remove this code block?

Reply:
The code means that parameters with defaults are coming through as nullable, it seemed redundant as the update to the types is also being done elsewhere, but I wasn't sure if there is a wider impact on other C# generators? Rather than removing this, an additional check could be added for default values (now that they are standard behaviour) - I don't think anyone would expect to get a nullable type for something which has a default.

Reply:
cc  @mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

cc @JFCote @devhl-labs

Reply:
A parameter with a default can be nullable. This block is something that should be refactored out in favor of using the isNullable property, but until that happens you should not remove it.